framework/src/modules/interoperability/mainchain/commands/recover_message.ts

```diff
@@ -182,20 +179,20 @@ export class RecoverMessageCommand extends BaseInteroperabilityCommand<Mainchain
 				ccm,
 				eventQueue: context.eventQueue.getChildQueue(ccmID),
 			};
+			let recoveredCCM: CCMsg;
 			// If the sending chain is the mainchain, recover the CCM.
 			// This function never raises an error.
 			if (ccm.sendingChainID.equals(getMainchainID(context.chainID))) {
-				await this._applyRecovery(ctx);
+				recoveredCCM = await this._applyRecovery(ctx);
 			} else {
 				// If the sending chain is not the mainchain, forward the CCM.
 				// This function never raises an error.
-				await this._forwardRecovery(ctx);
-
-				// Append the recovered CCM to the list of recovered CCMs.
-				// Notice that the ccm has been mutated in the applyRecovery and forwardRecovery functions
-				// as the status is set to CCM_STATUS_CODE_RECOVERED (so that it cannot be recovered again).
-				recoveredCCMs.push(crossChainMessage);
+				recoveredCCM = await this._forwardRecovery(ctx);
```

Comment:
@ricott1 just to make sure, when forwarding we also treat it as recovered?

Reply:
yes